### PR TITLE
Add documentation about usage of cv2eigen functions in eigen.hpp

### PR DESCRIPTION
**Pull request by themightyoarfish**
Addresses #17366 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable

### CODE REVIEW

**Comment by themightyoarfish** [comment]
It looks like this now:

Screenshot 2020-05-25 at 14 53 19

**Comment by asmorkalov** [comment]
Thanks for the contribution! This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first. We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no need to re-open PR, apply changes "inplace".

File: modules/core/include/opencv2/core/eigen.hpp [hunk 1]
```diff
@@ -53,6 +53,18 @@
 #pragma warning( disable: 4244 ) //conversion from '__int64' to 'int', possible loss of data
 #endif
 
+/**
+  @addtogroup core_eigen
```
Comment (alalek) [inline modules/core/include/opencv2/core/eigen.hpp:57]: > addtogroup

Should go into "cv" namespace, see addtogroup command below.

**Comment by asmorkalov** [comment]
@themightyoarfish Friendly reminder/

**Comment by asmorkalov** [comment]
@alalek I rebased the patch and fixed doxygen syntax issue. The patch is ready for merge.

File: modules/core/include/opencv2/core/eigen.hpp [hunk 2]
```diff
@@ -72,6 +72,14 @@ namespace cv
 #ifdef OPENCV_EIGEN_TENSOR_SUPPORT
 /** @brief Converts an Eigen::Tensor to a cv::Mat.
 
+ These functions are provided for OpenCV-Eigen interoperability. They convert `Mat`
```
Comment (alalek) [inline modules/core/include/opencv2/core/eigen.hpp:75]: Wrong place.
It must be a group documentation instead of single function's note under `#ifdef`

Comment (alalek) [reply modules/core/include/opencv2/core/eigen.hpp:74]: Still wrong.
http://pullrequest.opencv.org/buildbot/export/pr/17368/docs/d0/daf/group__core__eigen.html

**Comment by asmorkalov** [comment]
@alalek It works as expected now.

**Pull request merged**